homeassistant/components/hydrawise/sensor.py

```diff
@@ -21,22 +22,106 @@
 from .coordinator import HydrawiseDataUpdateCoordinator
 from .entity import HydrawiseEntity
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class HydrawiseSensorEntityDescription(SensorEntityDescription):
+    """Describes Hydrawise binary sensor."""
+
+    value_fn: Callable[[HydrawiseSensor], Any]
+
+
+def _get_zone_watering_time(sensor: HydrawiseSensor) -> int:
+    if (current_run := sensor.zone.scheduled_runs.current_run) is not None:
+        return int(current_run.remaining_time.total_seconds() / 60)
+    return 0
+
+
+def _get_zone_next_cycle(sensor: HydrawiseSensor) -> datetime | None:
+    if (next_run := sensor.zone.scheduled_runs.next_run) is not None:
+        return dt_util.as_utc(next_run.start_time)
+    return None
+
+
+def _get_zone_daily_active_water_use(sensor: HydrawiseSensor) -> float:
+    # water use for the zone
+    daily_water_summary = sensor.coordinator.data.daily_water_use[sensor.controller.id]
+    if sensor.zone_id is not None:  # redundant check needed for mypy
+        return float(daily_water_summary.active_use_by_zone_id.get(sensor.zone_id, 0.0))
+    return 0.0  # pragma: no cover
```

Comment:
```suggestion
    return float(daily_water_summary.active_use_by_zone_id.get(sensor.zone.id, 0.0))
```

I think this would work as well

Reply:
Fixed.